src/html5_parser/input_stream.rs

```diff
@@ -41,29 +46,36 @@ impl fmt::Display for Position {
     }
 }
 
+/// Defines a single character/element in the stream. This is either a UTF8 character, or
+/// a surrogate characters since these cannot be stored in a single char.
+/// Eof is denoted as a separate element.
 #[derive(PartialEq, Debug, Copy, Clone)]
 pub enum Element {
     /// Standard UTF character
     Utf8(char),
-    /// Surrogate character (since they cannot be stored in <char>)
+    /// Surrogate character (since they cannot be stored in char)
     Surrogate(u16),
     /// End of stream
     Eof,
 }
 
 impl Element {
+    /// Returns true when the element is an EOF
```

Comment:
```suggestion
    /// Whether the element is an EOF
```

You get the idea.  Similar suggestions apply further down.